Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -609,6 +609,379 @@ Start Safari on the iOS device. Run the above Python script on your connected ho
+iGoat-Swift                       0x100b7c000  2506752 (2.4 MiB)     /var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGo...
+TweakInject.dylib                 0x100eb4000  16384 (16.0 KiB)      /usr/lib/TweakInject.dylib
+SystemConfiguration               0x1862c0000  446464 (436.0 KiB)    /System/Library/Frameworks/SystemConfiguration.framework/SystemConfiguratio...
+libc++.1.dylib                    0x1847c0000  368640 (360.0 KiB)    /usr/lib/libc++.1.dylib
+```
+
+###### In-Memory Search
+
+In-memory search is a very useful technique to test for sensitive data that might be present in the app memory.
+
+See r2frida's help on the search command (`\/?`) to learn about the search command and get a list of options. The following shows only a subset of them:
+
+```bash
+[0x00000000]> \/?
+ /      search
+ /j     search json
+ /w     search wide
+ /wj    search wide json
+ /x     search hex
+ /xj    search hex json
+...
+```
+
+You can adjust your search by using the search settings `\e~search`. For example, `\e search.quiet=true;` will print only the results and hide search progress:
+
+```bash
+[0x00000000]> \e~search
+e search.in=perm:r--
+e search.quiet=false
+```
+
+For now, we'll continue with the defaults and concentrate on string search. In this first example, you can start by searching for something that you know it should be located in the main binary of the app:
+
+```bash
+[0x00000000]> \/ iGoat
+Searching 5 bytes: 69 47 6f 61 74
+Searching 5 bytes in [0x0000000100b7c000-0x0000000100de0000]
+...
+hits: 509
+0x100d7d332 hit2_0 iGoat_Swift24StringAnalysisExerciseVCC
+0x100d7d3b2 hit2_1 iGoat_Swift28BrokenCryptographyExerciseVCC
+0x100d7d442 hit2_2 iGoat_Swift23BackgroundingExerciseVCC
+0x100d7d4b2 hit2_3 iGoat_Swift9AboutCellC
+0x100d7d522 hit2_4 iGoat_Swift12FadeAnimatorV
+```
+
+Now take the first hit, seek to it and check your current location in the memory map:
+
+```bash
+[0x00000000]> s 0x100d7d332
+[0x100d7d332]> \dm.
+0x0000000100b7c000 - 0x0000000100de0000 r-x /private/var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGoat-Swift.app/iGoat-Swift
+```
+
+As expected, you are located in the region of the main iGoat-Swift binary (r-x, read and execute). In the previous section, you saw that the main binary is located between `0x0000000100b7c000` and `0x0000000100e97000`.
+
+Now, for this second example, you can search for something that's not in the app binary nor in any loaded library, typically user input. Open the iGoat-Swift app and navigate in the menu to Authentication -> Remote Authentication -> Start. There you'll find a password field that you can overwrite. Write the string "owasp-mstg" but do not click on Login just yet. You will perform two steps.
+
+```bash
+[0x00000000]> \/ owasp-mstg
+hits: 1
+0x1c06619c0 hit3_0 owasp-mstg
+
+[0x100d7d332]> s 0x1c06619c0
+[0x1c06619c0]> \dm.
+0x00000001c0000000 - 0x00000001c8000000 rw-
```

Comment:
these steps are not really clear to me...

Reply:
done